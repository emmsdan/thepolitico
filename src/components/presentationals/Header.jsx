import React from 'react';
import PropTypes from 'prop-types';
import Image from './Image';
import logo from '../../assets/images/official/logo.png';
import NavBar from './NavBar';

const image = {
  class: 'logo',
  src: logo,
  alt: 'official politico logo',
  style: '',
};
const pages = [
  { url: '/index', title: 'Home' },
  { url: '/office', title: 'office' },
  { url: '/election-result', title: 'result' },
  { url: '/auth', title: 'My Account' },
  {
    url: '/logout',
    title: 'Logout',
    style: 'background: none;',
    class: 'inactive hide',
  },
];

const Header = props => {
  return (
    <header className={props.header.url}>
      <a href="/">
        <Image image={image} />
      </a>
      <span className="show-nav"> &#9776; </span>
      <NavBar pages={pages} />
    </header>
  );
};

Header.propTypes = {
  header: PropTypes.shape({
    url: PropTypes.string,
    poweredby: PropTypes.shape({
      url: PropTypes.string,
      text: PropTypes.string,
    }),
  }),
};

export default Header;
