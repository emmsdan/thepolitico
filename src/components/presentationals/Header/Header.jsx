import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';
import logo from '../../assets/images/official/logo.png';
import NavBar from '../NavBar';

const Header = ({ pages }) => {
  return (
    <header>
      <a href="/">
        <Image className="logo" src={logo} alt="official politico logo" />
      </a>
      <span className="show-nav"> &#9776; </span>
      <NavBar pages={pages} />
    </header>
  );
};

Header.propTypes = {
  pages: PropTypes.any,
};

export default Header;
