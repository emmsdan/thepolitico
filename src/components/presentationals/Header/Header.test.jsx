import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('<Header />', () => {
  const header = [
    { url: '/index', title: 'Home' },
    { url: '/office', title: 'office' },
    { url: '/election-result', title: 'result' },
    { url: '/auth', title: 'My Account' },
  ];

  it('should render correctly', () => {
    const component = shallow(<Header page={header} />);
    expect(component).toMatchSnapshot();
  });
});
