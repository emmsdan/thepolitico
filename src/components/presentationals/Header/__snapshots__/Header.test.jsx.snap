// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header>
  <a
    href="/"
  >
    <Image
      alt="official politico logo"
      className="logo"
      src={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
  </a>
  <span
    className="show-nav"
  >
     â˜° 
  </span>
  <NavBar />
</header>
`;
